build/gen_stub.php

```diff
@@ -941,11 +1000,176 @@ public function getMethodSynopsisElement(array $funcMap, array $aliasMap, DOMDoc
+                } else {
+                    throw new Exception("Property $this->name has an unsupported union type");
+                }
+            } else {
+                $typeCode = "(zend_type) ZEND_TYPE_INIT_MASK(" . $arginfoType->toTypeMask() . ")";
+            }
+
+            $code .= $this->initializeValue($defaultValueType, $defaultValue);
+
+            if ($this->isKnownName) {
+                $nameCode = "ZSTR_KNOWN(ZEND_STR_" . strtoupper($propertyName) . ")";
+            } else {
+                $code .= "\tzend_string *property_{$propertyName}_name = zend_string_init(\"$propertyName\", sizeof(\"$propertyName\") - 1, 1);\n";
+                $nameCode = "property_{$propertyName}_name";
+            }
+
+            $code .= "\tzend_declare_typed_property(class_entry, $nameCode, &property_{$propertyName}_default_value, " . $this->getFlagsAsString() . ", NULL, $typeCode);\n";
+            if (!$this->isKnownName) {
+                $code .= "\tzend_string_release(property_{$propertyName}_name);\n";
+            }
+        } else {
+            if ($this->isKnownName) {
+                Throw new Exception("Non-typed property $this->name cannot be annotated as @known");
+            }
+
+            switch ($defaultValueType) {
+                case "undefined": // intentional fallthrough
+                case "NULL":
+                    $code .= "\tzend_declare_property_null(class_entry, \"$propertyName\", sizeof(\"$propertyName\") - 1, " . $this->getFlagsAsString() . ");\n";
+                    break;
+
+                case "boolean":
+                    $code .= "\tzend_declare_property_bool(class_entry, \"$propertyName\", sizeof(\"$propertyName\") - 1, " . ((int) $defaultValue) . ", " . $this->getFlagsAsString() . ");\n";
+                    break;
+
+                case "integer":
+                    $code .= "\tzend_declare_property_long(class_entry, \"$propertyName\", sizeof(\"$propertyName\") - 1, " . $defaultValue . ", " . $this->getFlagsAsString() . ");\n";
+                    break;
+
+                case "double":
+                    $code .= "\tzend_declare_property_double(class_entry, \"$propertyName\", sizeof(\"$propertyName\") - 1, " . $defaultValue . ", " . $this->getFlagsAsString() . ");\n";
+                    break;
+
+                case "string":
+                    $code .= "\tzend_declare_property_string(class_entry, \"$propertyName\", sizeof(\"$propertyName\") - 1, \"" . $defaultValue . "\", " . $this->getFlagsAsString() . ");\n";
+                    break;
+
+                case "array":
+                    $code .= $this->initializeValue($defaultValueType, $defaultValue);
+                    $code .= "\tzend_declare_property_ex(class_entry, \"$propertyName\", sizeof(\"$propertyName\") - 1, property_{$propertyName}_default_value, " . $this->getFlagsAsString() . ");\n";
```

Comment:
Might make sense to just always use declare_property_ex, instead of generating more specific code?

Reply:
Hmm, I'm not sure. I like that the generated code is more concise this way. This is not a hard preference, so I can change the implementation if you see more advantages of using `zend_declare_property_ex`.

Reply:
I don't think conciseness of generated code is particularly important -- especially if we're going to be primarily using typed properties going forward anyway :)

Reply:
I've just pushed a change which generates `declare_property_ex ` calls :)